### PR TITLE
Kubelet changes to support vertical scaling

### PR DESCRIPTION
**Pull request by vinaykul**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #

**Special notes for your reviewer**:

**Release note**:

```release-note

```

### CODE REVIEW

File: pkg/kubelet/kubelet.go [hunk 1]
```diff
@@ -1564,10 +1564,10 @@ func (kl *Kubelet) syncPod(o syncPodOptions) error {
 				if err := kl.containerManager.UpdateQOSCgroups(); err != nil {
 					glog.V(2).Infof("Failed to update QoS cgroups while syncing pod: %v", err)
 				}
```
Comment (pdgetrf) [inline pkg/kubelet/kubelet.go]: this change shouldn't belong to this PR

File: pkg/kubelet/kuberuntime/kuberuntime_container.go [hunk 2]
```diff
@@ -614,6 +616,54 @@ func (m *kubeGenericRuntimeManager) killContainersWithSyncResult(pod *v1.Pod, ru
 	return
 }
 
+func (m *kubeGenericRuntimeManager) updateContainer(pod *v1.Pod, containerID kubecontainer.ContainerID, containerName string, resources v1.ResourceRequirements) error {
```
Comment (pdgetrf) [inline pkg/kubelet/kuberuntime/kuberuntime_container.go]: minor: name to updateContainerResources(...)

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go [hunk 3]
```diff
@@ -71,6 +72,56 @@ func TestRemoveContainer(t *testing.T) {
...
+	containerId := fakeContainers[0].Id
+
+	err = m.updateContainer(testPod, testContainerID, testContainerName, testResources)
+	assert.NoError(t, err)
+
+	// Verify container is running
+	containers, err := fakeRuntime.ListContainers(&runtimeapi.ContainerFilter{Id: containerId})
+	assert.NoError(t, err)
+	assert.Equal(t, len(containers), 1)
+}
```
Comment (pdgetrf) [inline pkg/kubelet/kuberuntime/kuberuntime_container_test.go:123]: is it possible to verify resource has actually been updated?

**Review by pdgetrf — APPROVED** [review]
see minor suggestions inline

File: pkg/kubelet/kuberuntime/kuberuntime_container_test.go (hunk 3 above)
Comment (vinaykul) [reply pkg/kubelet/kuberuntime/kuberuntime_container_test.go:123]: It should be. I have to dig into the fake-runtime more and see.

File: pkg/kubelet/kubelet.go (hunk 1 above)
Comment (vinaykul) [reply pkg/kubelet/kubelet.go]: I'll remove it. It was in the POC but I wasn't sure where this came from - dont remember if I made this change. Is this your change?

**Pull request merged**